File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/RedSlaver_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class RedSlaver_Model extends MonsterInfo{
    public static Id = "RedSlaver"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(48, 52);
        else
            return Mathf.RandomRange(46, 50);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/redSlaver/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/redSlaver/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/redSlaver/skeleton.png"));
    }

    get Name(): string {
        return "奴隶贩子";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/RedSlaver_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class RedSlaver_Model : MonsterInfo
    {
        public static string Id = "RedSlaver";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(48, 52) : Random.Range(46,50);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/redSlaver/skeleton.atlas",
                    "Monsters/theBottom/redSlaver/skeleton.json", "Monsters/theBottom/redSlaver/skeleton.png");
            }
        }

        public override string Name => "奴隶贩子";

        public RedSlaver_Model(Vector2 pos) : base(pos)
        {
        }
    }
}